io.go

```diff
@@ -6,16 +6,38 @@ import (
 	"io"
 )
 
-func Copy(ch chan<- []byte, r io.Reader) error {
+func Copy(ch chan<- []byte, r io.Reader, w io.Writer) error {
 	scanner := bufio.NewScanner(r)
 
 	for scanner.Scan() {
-		ch <- []byte(scanner.Text() + "\n")
+		line := scanner.Bytes()
+		cpy := make([]byte, len(line)+1)
+		copy(cpy, line)
+		line = append(line, '\n')
+
+		if err := writeFully(w, line); err != nil {
+			return err
+		}
+
+		ch <- []byte(line)
 	}
 
 	return scanner.Err()
 }
 
+func writeFully(w io.Writer, bytes []byte) error {
```

Comment:
So was I! And in practice `w.Write` writes the entire argument often enough that I considered pretending it's defined to do so (even though it's not) for simplicity's sake.